rest/oidc_api_test.go

```diff
@@ -758,3 +806,113 @@ func refreshProviderConfig(providers auth.OIDCProviderMap, issuer string) {
+			},
+		}, {
+			name: "successful registered user authentication against single provider",
+			providers: auth.OIDCProviderMap{
+				"foo": mockProvider("foo"),
+			},
+			defaultProvider:     "foo",
+			authURL:             "/db/_oidc?provider=foo&offline=true",
+			requireExistingUser: true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			mockAuthServer, err := newMockAuthServer()
+			require.NoError(t, err, "Error creating mock oauth2 server")
+			mockAuthServer.Start()
+			defer mockAuthServer.Shutdown()
+			mockAuthServer.options.issuer = mockAuthServer.URL + "/" + tc.defaultProvider
+			refreshProviderConfig(tc.providers, mockAuthServer.URL)
+
+			opts := auth.OIDCOptions{Providers: tc.providers, DefaultProvider: &tc.defaultProvider}
+			restTesterConfig := RestTesterConfig{DatabaseConfig: &DbConfig{OIDCConfig: &opts}}
+			restTester := NewRestTester(t, &restTesterConfig)
+			defer restTester.Close()
+
+			// Create the user first if the test requires a registered user.
+			if tc.requireExistingUser {
+				body := `{"name":"foo_noah", "password":"pass", "admin_channels":["foo"]}`
+				userResponse := restTester.SendAdminRequest(http.MethodPut, "/db/_user/foo_noah", body)
+				assertStatus(t, userResponse, http.StatusCreated)
+				userResponse = restTester.SendAdminRequest(http.MethodGet, "/db/_user/foo_noah", "")
+				assertStatus(t, userResponse, http.StatusOK)
+			}
+
+			mockSyncGateway := httptest.NewServer(restTester.TestPublicHandler())
+			defer mockSyncGateway.Close()
+			mockSyncGatewayURL := mockSyncGateway.URL
+
+			token, err := mockAuthServer.makeToken()
+			require.NoError(t, err, "Error obtaining signed token from OpenID Connect provider")
+			require.NotEmpty(t, token, "Empty token retrieved from OpenID Connect provider")
+			sessionEndpoint := mockSyncGatewayURL + "/" + restTester.DatabaseConfig.Name + "/_session"
+
+			request, err := http.NewRequest(http.MethodPost, sessionEndpoint, strings.NewReader(`{}`))
+			require.NoError(t, err, "Error creating new request")
+			request.Header.Add("Authorization", BearerToken+" "+token)
+			response, err := http.DefaultClient.Do(request)
+			require.NoError(t, err, "Error sending request with bearer token")
+
+			if (forceError{}) != tc.expectedError {
+				assertHttpResponse(t, response, tc.expectedError)
+				return
+			}
+			require.Equal(t, http.StatusOK, response.StatusCode)
+			assert.Equal(t, "application/json", response.Header.Get("Content-Type"))
+
+			var responseBody map[string]interface{}
+			require.NoError(t, err, json.NewDecoder(response.Body).Decode(&responseBody))
+			sessionCookie := getCookie(response.Cookies(), auth.DefaultCookieName)
+			require.NotNil(t, sessionCookie, "No session cookie found")
+			sessionExpiry := int(sessionCookie.Expires.Sub(time.Now()).Seconds())
+			assert.True(t, 86395 <= sessionExpiry && 86400 >= sessionExpiry, "session expiry is not within 5 seconds of 24 hours")
```

Comment:
24h is the default SG session timeout (kDefaultSessionTTL).   Is that what we want to be testing here?  It's a bit unexpected in  what appears to be an OIDC test.